### PR TITLE
Branch enum tag

### PR DESCRIPTION
**Pull request by tanteckfang**
Added Enum class TagType to distinguish the type of tags.

### CODE REVIEW

File: src/main/java/seedu/address/model/tag/TagType.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        return displayName;
+    }
+
+    /**
+     * Method to transform a string to a Tagtype.
+     *
+     * @param name name of the String.
+     * @return Tagtype of the name passed in.
+     */
+    public static TagType fromString(String name) {
```
Comment (joel-foo) [inline src/main/java/seedu/address/model/tag/TagType.java:32]: i think it's better to include "throws IlegalArgumentException" in the function header

**Review by joel-foo — CHANGES_REQUESTED** [review]
apart from the requested change, LGTM!

File: src/main/java/seedu/address/model/tag/TagType.java (hunk 1 above)
Comment (tanteckfang) [reply src/main/java/seedu/address/model/tag/TagType.java]: Hi Joel, thanks for pointing out this issue, I have edited such that it includes the throws execption.

**Review by joel-foo — APPROVED** [review]
LGTM!

**Pull request merged**